clouddriver-oracle-bmcs/src/main/groovy/com/netflix/spinnaker/clouddriver/oraclebmcs/provider/view/OracleBMCSInstanceProvider.groovy

```diff
@@ -52,7 +52,7 @@ class OracleBMCSInstanceProvider implements InstanceProvider<OracleBMCSInstance>
       log.warn("There should be at most one identifier!")
     }
     Set<OracleBMCSInstance> instances = loadInstances(identifiers)
-    return instances.first()
+    return instances?.isEmpty() ? null: instances.first()
```

Comment:
The problem is that `[].first()` will still throw a NoSuchElementException, which is what i'm trying to safeguard against. It is a shame IMO that the .first() groovy addon method doesn't just return null when the collection is empty.